lib/Dialect/FIRRTL/Transforms/EmitOMIR.cpp

```diff
@@ -1208,6 +1244,51 @@ hw::InnerRefAttr EmitOMIRPass::getInnerRefTo(FModuleLike module,
                          });
 }
 
+FIRRTLType EmitOMIRPass::getTypeOf(Operation *op) {
+  assert(op->getNumResults() == 1 &&
+         op->getResult(0).getType().isa<FIRRTLType>() &&
+         "op must have a single FIRRTLType result");
+  return op->getResult(0).getType().cast<FIRRTLType>();
+}
+
+FIRRTLType EmitOMIRPass::getTypeOf(FModuleLike mod, size_t portIdx) {
+  Type portType = mod.getPortType(portIdx);
+  assert(portType.isa<FIRRTLType>() && "port must have a FIRRTLType");
+  return portType.cast<FIRRTLType>();
+}
+
+// Returns true if the tracker has a non-zero field ID.
+bool EmitOMIRPass::hasFieldID(Tracker tracker) { return tracker.fieldID > 0; }
+
+// Constructs a reference to a field from a FIRRTLType with a fieldID.
+SmallString<8> EmitOMIRPass::addFieldID(FIRRTLType type, unsigned fieldID) {
```

Comment:
Passing the `SmallString` by reference and then manipulating it directly would likely be better than (I think) returning a copy of it. It also might work to make the thing passed in a `SmallVectorImpl<char>` to avoid making this an explicit `SmallString<8>`. Not sure about that, though.